module-bsp/devices/temperature/CT7117.cpp

```diff
@@ -0,0 +1,145 @@
+                             const std::uint16_t reg)
+    {
+        const drivers::I2CAddress addr = {
+            .deviceAddress = dev_id, .subAddress = static_cast<std::uint32_t>(address), .subAddressSize = 1};
+
+        return i2c.Write(addr, &to_bytes(reg)[0], sizeof(reg));
+    }
+
+    std::optional<std::uint16_t> read_register16(const std::uint8_t dev_id,
+                                                 drivers::DriverI2C &i2c,
+                                                 const Registers address)
+    {
+        const drivers::I2CAddress addr = {
+            .deviceAddress = dev_id, .subAddress = static_cast<std::uint32_t>(address), .subAddressSize = 1};
+        std::array<std::uint8_t, 2U> ret_value{};
+
+        if (const auto result = i2c.Read(addr, &ret_value[0], ret_value.size()); result == ret_value.size()) {
+            return from_bytes(ret_value);
+        }
+        return std::nullopt;
+    }
+
+    units::Temperature from_raw(const std::uint16_t raw)
+    {
+        if ((raw & 0x8000) != 0) { // sign bit present
+            const auto integer =
+                static_cast<std::uint16_t>(((~raw + 1) & 0x7FFF) >> 7); // remove sign bit and shift to lower byte
+            const auto fractional = static_cast<std::uint16_t>(((~raw + 1) & 0x7F) * 0.78125);
+            return -1 * (static_cast<float>(integer) + (static_cast<float>(fractional) / 100.0));
+        }
+        else {
+            const auto integer =
+                static_cast<std::uint16_t>((raw & 0x7FFF) >> 7); // remove sign bit and shift to lower byte
+            const auto fractional = static_cast<std::uint16_t>((raw & 0x7F) * 0.78125);
+            return static_cast<float>(integer) + (static_cast<float>(fractional) / 100.0);
+        }
+    }
+} // namespace
+
+namespace bsp::devices::temperature::CT7117
+{
+
+    CT7117::CT7117(const std::uint8_t id, drivers::DriverI2C &i2c) : device_id{id}, i2c{i2c}
+    {}
+
+    CT7117::~CT7117()
+    {
+        standby();
+    }
+
+    bool CT7117::standby()
+    {
+        auto reg = read_register16(device_id, i2c, Registers::Config);
+        if (not reg) {
+            return false;
+        }
+        *reg |= static_cast<std::uint16_t>(ConfigReg::SD);
+        return write_register16(device_id, i2c, Registers::Config, *reg) == sizeof(*reg);
+    }
+
+    bool CT7117::wakeup()
+    {
+        auto reg = read_register16(device_id, i2c, Registers::Config);
+        if (not reg) {
+            return false;
+        }
+        *reg &= ~(static_cast<std::uint16_t>(ConfigReg::SD));
+        return write_register16(device_id, i2c, Registers::Config, *reg) == sizeof(*reg);
+    }
+
+    std::optional<units::Temperature> CT7117::get_temperature() const
+    {
+        auto reg = read_register16(device_id, i2c, Registers::Temp);
+        if (not reg) {
+            return std::nullopt;
+        }
+
+        /// Limit resolution to 0.25°C
+        *reg &= 0xFFE0;
```

Comment:
Code was copier only from another file but here could be  added variable named quaterCelciusDegreeResolution(or similar) instead of comment